README.md

```diff
@@ -68,25 +68,54 @@ On **unsuccessful** verification of the given pass, the `verifyPassURI` method r
 }
 ```
 
-### Advanced Parameters
 
-To allow for flexibility in this library, `verifyPassURIWithTrustedIssuers` method allows for additional parameters as documented below.
+### Verifying example COVID Passes
 
-#### Custom Trusted Issuers
+To verify [example COVID Passes from the spec](https://nzcp.covid19.health.nz/#valid-worked-example), use `verifyPassURIWithTrustedIssuers` with the following parameters:
 
 ```javascript
 import { verifyPassURIWithTrustedIssuers } from "@vaxxnz/nzcp";
 
-// An array of trusted issuers which work with the NZ COVID Pass - Technical Specification
-// https://nzcp.covid19.health.nz/
-const nzcpTrustedIssuers = ["did:web:nzcp.covid19.health.nz"];
+// An array of trusted issuers for the example COVID Passes
+const exampleTrustedIssuers = ["did:web:nzcp.covid19.health.nz"];
 
 const result = await verifyPassURIWithTrustedIssuers(
-  "NZCP:/1/2KCEVIQEIVVWK6...", // COVID-19 Pass to be verified
-  nzcpTrustedIssuers // Array of trusted issuers
+  "NZCP:/1/2KCEVIQEIVVWK6...",  // COVID-19 Pass to be verified
+  exampleTrustedIssuers         // Array of trusted issuers
 );
 ```
 
+### Offline support
+
+For offline verification, you can use the `verifyPassURI` method with the following parameters:
+
+```javascript
+import { verifyPassURI } from "@vaxxnz/nzcp";
+import liveDIDDocument from "./liveDIDDocument.json"
+
+const result = await verifyPassURI(
+  "NZCP:/1/2KCEVIQEIVVWK6...",
+  [liveDIDDocument]             // Array of DID documents accessible offline
+);
```

Comment:
My only feedback is on this API design.

It is going to be a bit annoying to work on this unless we bite the bullet now and make a backwards incompatible changes.

I propose we 

1. Kill verifyPassURIWithTrustedIssuers
2. Change the signatures of these functions
3. Export our own version of the MoH DID_DOCUMENTS and TRUSTED_ISSUERS

```
import { verifyPassUri, verifyPassUriOffline, DID_DOCUMENTS, TRUSTED_ISSUERS } from "@vaxxnz/nzcp-js"

// Standard usage, resolves DID document, resolves according to spec
const result = await verifyPassUri("NZCP:/....");

// Standard usage, resolves DID document (same as previous call
const result = await verifyPassUri("NZCP:/....", { trustedIssuer: TRUSTED_ISSUERS.MOH_PROD });

// Standard usage, resolves DID document, use MoH test issuer
const result = await verifyPassUri("NZCP:/....", { trustedIssuer: TRUSTED_ISSUERS.MOH_TEST });

// Standard usage, resolves DID document, pass your own trusted issuers
const result = await verifyPassUri("NZCP:/....", { trustedIssuer: myTrustedIssuer });

// Standard usage, resolves DID document, allowed trustedIssuer to be string|string[]
const result = await verifyPassUri("NZCP:/....", { trustedIssuer: [myTrustedIssuer] });

// offline usage, use hard coded DID document
const result = verifyPassUriOffline("NZCP:/...");

// offline usage, use hard coded DID document (same as the previous call)
const result = verifyPassUriOffline("NZCP:/...", { didDocument: DID_DOCUMENTS.MOH_PROD });

// offline usage, use hard coded DID document, use MoH test DID document
const result = verifyPassUriOffline("NZCP:/...", { didDocument: DID_DOCUMENTS.MOH_TEST });

// offline usage, pass your own DID document (needs a different interface since it won't resolve a promise)
const result = verifyPassUriOffline("NZCP:/...", { didDocument: myDidDocument });

// offline usage, pass your own DID document, allow didDocument to be string|string[]
const result = verifyPassUriOffline("NZCP:/...", { didDocument: [myDidDocument] });
```

Reply:
Also the idea is lets just change the second param to be an `options` object so we can add changes or other params easily

Reply:
if we break compatibility, then we could do:

```typescript
await verifyPassURI(uri: string, { trustedIssuers?: string[], didDocuments?: DIDDocument[] })
```

- having trustedIssuers sets the trusted issuers to those specific trusted issuers
- having didDocuments means offline mode

Reply:
I don't think we should expose `DID_DOCUMENTS.MOH_TEST` and `DID_DOCUMENTS.MOH_PROD` since it's not in our prerogative to vendor them. I think we should rather have a function `downloadDidDocuments(didUrls: sting[])`.

I think we can add it in next release when the CORS issue is fixed.

Reply:
Will convinced me to go with the API above. I've written the code for it but need to test.